eng/pipelines/templates/jobs/ci.yml

```diff
@@ -75,10 +78,13 @@ jobs:
           arguments: '--set-pipeline-variable CheckoutDirectories --project-list $(ProjectList)'
           workingDirectory: '$(System.DefaultWorkingDirectory)'
 
-      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
-        parameters:
-          Paths: $(CheckoutDirectories)
-          SkipDefaultCheckout: true
+      # Skip sparse checkout for the `azure-sdk-for-<lang>-pr` private mirrored repositories
+      # as we require the github service connection to be loaded.
+      - ${{ if not(contains(variables['Build.DefinitionName'], 'java-pr')) }}:
```

Comment:
I think `not(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-java-pr')` willbe better to use.